File: Tests/ErrorTests.m
//
//  ErrorTests.m
//  RaptureXML
//
//  Created by John Blanco on 9/24/11.
//  Copyright (c) 2011 Rapture In Venice. All rights reserved.
//

#import <XCTest/XCTest.h>
#import "RXMLElement.h"

@interface ErrorTests : XCTestCase {
    NSString *simplifiedXML_;
    NSString *badXML_;
}

@end



@implementation ErrorTests

- (void)setUp {
    simplifiedXML_ = @"\
        <shapes>\
            <square>Square</square>\
            <triangle>Triangle</triangle>\
            <circle>Circle</circle>\
        </shapes>";
    badXML_ = @"</xml";
}

- (void)testBadXML {
    RXMLElement *rxml = [RXMLElement elementFromXMLString:badXML_ encoding:NSUTF8StringEncoding];
    XCTAssertFalse([rxml isValid]);
}

- (void)testMissingTag {
    RXMLElement *rxml = [RXMLElement elementFromXMLString:simplifiedXML_ encoding:NSUTF8StringEncoding];
    RXMLElement *hexagon = [rxml child:@"hexagon"];
    
    XCTAssertNil(hexagon);
}

- (void)testMissingTagIteration {
    RXMLElement *rxml = [RXMLElement elementFromXMLString:simplifiedXML_ encoding:NSUTF8StringEncoding];
    __block NSInteger i = 0;
    
    [rxml iterate:@"hexagon" usingBlock:^(RXMLElement *e) {
        i++;
    }];
     
    XCTAssertEqual(i, 0);
}

- (void)testMissingAttribute {
    RXMLElement *rxml = [RXMLElement elementFromXMLString:simplifiedXML_ encoding:NSUTF8StringEncoding];
    NSString *missingName = [rxml attribute:@"name"];
    
    XCTAssertNil(missingName);
}

@end

File: Swift/RaptureXMLTests/ErrorTests.swift
//
//  ErrorTests.swift
//  RaptureXML-Swift
//
//  Created by Brett Lamy on 2/12/15.
//  Copyright (c) 2015 Brett Lamy. All rights reserved.
//

import UIKit
import XCTest

class ErrorTests: XCTestCase {
    var simplifiedXML_ = "" +
    "<shapes>" +
    "<square>Square</square>" +
    "<triangle>Triangle</triangle>" +
    "<circle>Circle</circle>" +
    "</shapes>";
    var badXML_ = "</xml";
   
    func testBadXML() {
        let rxml = RXMLElement(fromXMLString:badXML_, encoding: NSUTF8StringEncoding);
        XCTAssertFalse(rxml.isValid);
    }

    func testMissingTag() {
        let rxml = RXMLElement(fromXMLString:simplifiedXML_, encoding: NSUTF8StringEncoding);
        let hexagon = rxml.child("hexagon");
        XCTAssertNil(hexagon);
    }
    
    func testMissingTagIteration() {
        let rxml = RXMLElement(fromXMLString:simplifiedXML_, encoding: NSUTF8StringEncoding);
        var i = 0;
        
        rxml.iterate("hexagon", usingBlock: { (e : RXMLElement!) -> Void in
            i+=1;
        })
        
        XCTAssertEqual(i, 0);
    }
    
    func testMissingAttribute() {
        let rxml = RXMLElement(fromXMLString:simplifiedXML_, encoding: NSUTF8StringEncoding);
        var missingName = rxml.attribute("name");
        
        XCTAssertNil(missingName);
    }
    

}
